### PR TITLE
Added limits to nested calls and in-clause.

### PR DESCRIPTION
**Pull request by idralyuk**

### CODE REVIEW

File: modules/engine/lib/engine/select.js [hunk 1]
```diff
@@ -66,6 +66,9 @@ exports.exec = function(opts, statement, cb, parentEvent) {
                 }(cloned));
             });
 
+            // Determine whether the number of funcs is within the limit and prune the funcs array
+            funcs = funcs.slice(0, getMaxNestedCalls(opts));
```
Comment (s3u) [inline modules/engine/lib/engine/select.js]: The call to getMax... should be done just once. I would declare 

```
var max;
```

before exports.exc(...) and add

```
max = max || getMaxNestedCalls(opts);
```

I would also rename getMaxNestedCalls to getMaxNestedRequests.

File: modules/engine/lib/engine/select.js [hunk 2]
```diff
@@ -314,3 +321,17 @@ var clone = function(obj) {
     return temp;
 };
 
+function getMaxNestedCalls(opts) {
+    var maxNestedCalls, config = opts.config;
+
+    if (config && config.ebay && config.ebay.maxNestedCalls) {
```
Comment (s3u) [inline modules/engine/lib/engine/select.js]: Why "ebay"?

**Comment by s3u** [comment]
The setting should just be global, with a systemic default that can be overridden in config.

**Comment by idralyuk** [comment]
Made the requested modifications. Original thinking was to support dynamic reloading of the property.

**Comment by s3u** [comment]
I'm merging this, but could you emit a log when funcs is sliced to the max?

**Comment by idralyuk** [comment]
Thanks. Done. See https://github.com/ql-io/ql.io/pull/215

**Pull request merged**